mm2src/coins/qrc20.rs

```diff
@@ -416,6 +432,69 @@ impl Qrc20Coin {
         })
     }
 
+    async fn add_delegation(
+        &self,
+        amount: BigDecimal,
+        to_addr: Address,
+        fee: u64,
+    ) -> Result<GenerateQrc20TxResult, MmError<GenerateTxError>> {
+        let staker_address_hex = qtum::contract_addr_from_utxo_addr(to_addr.clone());
+        let delegation_output = self.add_delegation_output(
+            staker_address_hex,
+            fee,
+            QRC20_GAS_LIMIT_DELEGATION,
+            QRC20_GAS_PRICE_DEFAULT,
+        )?;
+
+        //  change_script = CScript([OP_DUP, OP_HASH160, bytes.fromhex(delegator_address_hex), OP_EQUALVERIFY, OP_CHECKSIG])
+        let change_script = ScriptBuilder::build_p2pkh(&to_addr.hash);
+
+        let amount_sat = sat_from_big_decimal(&amount, self.decimals()).unwrap()
+            - (QRC20_GAS_LIMIT_DELEGATION * QRC20_GAS_PRICE_DEFAULT);
+        let change_script_output = ContractCallOutput {
+            value: amount_sat,
+            script_pubkey: change_script.to_bytes(),
+            gas_limit: 0,
+            gas_price: 0,
+        };
+        let outputs = vec![delegation_output, change_script_output];
+        let _utxo_lock = UTXO_LOCK.lock().await;
+        let res = self.generate_qrc20_transaction(outputs).await?;
+        Ok(res)
+    }
+
+    pub fn add_delegation_output(
+        &self,
+        to_addr: H160,
+        fee: u64,
+        gas_limit: u64,
+        gas_price: u64,
+    ) -> Qrc20AbiResult<ContractCallOutput> {
+        let function: &ethabi::Function = QTUM_DELEGATE_CONTRACT.function("addDelegation")?;
+        let msg = b"\x15Qtum signed message:\n\x28";
+        let buffer: Vec<u8> = [msg.to_vec(), to_addr.to_vec()].concat();
+        let hashed = dhash256(&buffer);
+        let signature = self.utxo.key_pair.private().sign(&hashed)?;
+        let params = function.encode_input(&[
+            Token::Address(to_addr),
+            Token::Uint(fee.into()),
+            Token::Bytes(signature.into()),
+        ])?;
+
+        let contract_address = ethabi::Address::from_str("0000000000000000000000000000000000000086").unwrap();
```

Comment:
Should I turn it into a constant? It's the smart contract address of the delegation for QTUM.